### ISSUE TITLE
yaml2ck fails for phases with no reactions

### ISSUE THREAD

**Issue by speth** [Cantera/cantera#1679]
**Problem description**

Trying to convert a YAML mechanism to CK format fails when the phase has no reactions.

**Steps to reproduce**

1. Download attached `org.yaml.txt` and save as `org.yaml`
2. From a command prompt, run `yaml2ck --mechanism=chem.inp --thermo=therm.dat --overwrite org.yaml`

**Behavior**

`yaml2ck` fails with the following error:

```pytb
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "R:\ray\mamba\ctdev312\Lib\site-packages\cantera\yaml2ck.py", line 856, in <module>
    main()
  File "R:\ray\mamba\ctdev312\Lib\site-packages\cantera\yaml2ck.py", line 807, in main
    output_paths = convert(
                   ^^^^^^^^
  File "R:\ray\mamba\ctdev312\Lib\site-packages\cantera\yaml2ck.py", line 704, in convert
    mechanism_text.append(build_reactions_text(all_reactions))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "R:\ray\mamba\ctdev312\Lib\site-packages\cantera\yaml2ck.py", line 356, in build_reactions_text
    max_reaction_length = max(len(r.equation) for r in reactions)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: max() iterable argument is empty
```

**System information**

- Cantera version: 3.0.0 or `main` branch at 64ed2b732
- OS: Windows 11
- Python/MATLAB/other software versions: Python 3.12 (conda-forge)

**Attachments**

[org.yaml.txt](https://github.com/Cantera/cantera/files/14629510/org.yaml.txt)

**Additional context**

Originally reported on the [Cantera Users' Group](https://groups.google.com/g/cantera-users/c/pax4W_t_qpU)

**Comment by corykinney** [comment]
@speth would the desired behavior be to simply skip the `build_reactions_text` call if `n_reactions` (or in this case just `len(all_reactions)`) for the phase is zero? If so, I can implement a quick fix.

**Comment by speth** [comment]
Yeah, that sounds like it would be sufficient. Or I guess, the most Pythonic check would be `if all_reactions: ...`.

I just noticed the comment that @bryanwweber had put in where this is called:
```py
# TODO: Handle phases without reactions
```

**Comment by bryanwweber** [comment]
And this is why TODO comments aren't helpful 🤣🤣🤣

**Comment by corykinney** [comment]
I need to go ahead and add a test case. I've already confirmed the fix works for roundtripping with `ck2yaml`, but does Chemkin require a `REACTIONS` section to be present or is it okay if it's missing? The barebones example I tested with the fix referenced gives a reactions file like so:

```

ELEM
H
END

SPECIES
H2
END

```

If Chemkin requires it, I could modify `build_reactions_text` to at least do:

```
REACTIONS
END
```

but I figured I'd ask since I'm only somewhat familiar with Chemkin's file requirements.
